src/impl/realm_coordinator.hpp

```diff
@@ -172,6 +183,11 @@ class RealmCoordinator : public std::enable_shared_from_this<RealmCoordinator> {
 
     std::shared_ptr<SyncSession> m_sync_session;
 
+    std::mutex m_partial_sync_queue_mutex;
+    std::deque<std::tuple<std::string, std::string, std::string, ResultsNotifier&>> m_partial_sync_queue;
```

Comment:
Rather than using a tuple containing three strings that could easily be confused. It'd be clearer to use a struct with named fields.